helper/src/fs.rs

```diff
@@ -178,6 +183,11 @@ impl_path_lazylock! {
     "txpool",
 }
 
+/// Joins the [`Path`] with a folder for the given [`Network`].
+pub fn path_with_network(path: &Path, network: Network) -> PathBuf {
+    path.join(network.to_string())
+}
```

Comment:
The tree should be:
```bash
/home/alice/.cache/cuprate/{mainnet,testnet,stagenet}/*
```
This function and the way it's used will create:
```bash
/home/alice/.cache/cuprate/*/{mainnet,testnet,stagenet}/*
```

Maybe integrate `{mainnet,testnet,stagenet}` into the macro generating the PATHs so we have: `CUPRATE_{MAINNET,TESTNET,STAGENET}_{DATA,BLOCKCHAIN,CONFIG,...}_DIR`?

That's kinda hacky too but this function is a lot worse since it seeps into user space. `monerod` dumps `mainnet` into `.bitmonero/` and creates `.bitmonero/{testnet,stagenet}`. Maybe see how `bitcoin/eth/zcash` do it?

Reply:
So initially I wanted to do that, however it gets annoying when you allow users to specify custom paths for different things.

I think it might be best to remove the blockchain & txpool paths from the macro and instead have functions that build them on the fly from a `Network` and a `data_dir`, i.e.:

```rust
pub fn blockchain_path(data_dir: &Path, network: Network) -> PathBuf {
    path.join(network.to_string()).join("blockchain")
}
```

(I'll  change it so we match on the network and don't write `mainnet`).

Doing it like that makes it easier IMO rather than needing to match on the network to choose the right constant and potentially using a completely custom dir anyway.